web_src/js/utils/color.js

```diff
@@ -0,0 +1,53 @@
+// Check similar implementation in modules/util/color.go and keep synchronization
+// Return R, G, B values defined in reletive luminance
+function getLuminanceRGB(channel) {
+  const sRGB = channel / 255;
+  const res = (sRGB <= 0.03928) ? sRGB / 12.92 : ((sRGB + 0.055) / 1.055) ** 2.4;
+  return res;
+}
+
+// Reference from: https://www.w3.org/WAI/GL/wiki/Relative_luminance
+function getLuminance(r, g, b) {
+  const R = getLuminanceRGB(r);
+  const G = getLuminanceRGB(g);
+  const B = getLuminanceRGB(b);
+  const luminance = 0.2126 * R + 0.7152 * G + 0.0722 * B;
+  return luminance;
+}
+
+const convertHexUnitTo256 = (hexStr) => parseInt(hexStr.repeat(2 / hexStr.length), 16);
+
+const getAlphafloat = (a) => {
+  if (a !== undefined) {
+    return a / 255;
+  }
+  return 1;
+};
```

Comment:
```suggestion
function getAlphafloat(alpha) {
  if (alpha !== undefined) {
    return alpha / 255;
  }
  return 1;
}
```

prefer function declaration.